main/main.cpp

```diff
@@ -1801,12 +1801,22 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 		} else if (arg == "--" || arg == "++") {
 			adding_user_args = true;
 		} else {
-			if (!FileAccess::exists(arg) && !DirAccess::exists(arg)) {
+			// Handle arguments specific to C#-enabled builds.
+			const bool using_mono_args = arg == "--generate-mono-glue";
+#ifndef MODULE_MONO_ENABLED
+			if (using_mono_args) {
+				OS::get_singleton()->print(
+						"`--generate-mono-glue` was specified on the command line, but this Godot binary was compiled without .NET support. Aborting.\n"
+						"To be able to generate C# glue code, use the `module_mono_enabled=yes` SCons option when compiling Godot.\n");
+				goto error;
+			}
+#endif
```

Comment:
We're starting to accumulate a lot of hacks in `main.cpp` for the Mono module... I'm not fond of increasing the tech debt further here.

There's also `--class-db-json` in the Mono module that's not being handled here.

Reply:
See also #99224.

Reply:
> There's also `--class-db-json` in the Mono module that's not being handled here.

I don't think anybody's using this, it's an undocumented option and could probably be replaced with `--dump-extension-api`.